pkg/apis/messaging/v1/subscription_validation.go

```diff
@@ -21,17 +21,29 @@ import (
 
 	"github.com/google/go-cmp/cmp/cmpopts"
 	"k8s.io/apimachinery/pkg/api/equality"
+	"knative.dev/eventing/pkg/apis/feature"
+	cn "knative.dev/eventing/pkg/crossnamespace"
 	"knative.dev/pkg/apis"
 	duckv1 "knative.dev/pkg/apis/duck/v1"
 	"knative.dev/pkg/kmp"
 )
 
 func (s *Subscription) Validate(ctx context.Context) *apis.FieldError {
+	// s.Validate(ctx) because krshaped is defined on the entire subscription, not just the spec
```

Comment:
I guess this comment applies to the `if feature.FromContext(ctx).IsEnabled(feature.CrossNamespaceEventLinks) { ...` code section, should it be placed before line 38 ?